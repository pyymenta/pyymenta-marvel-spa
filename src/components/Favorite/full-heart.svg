<?xml version="1.0" encoding="UTF-8"?>
<svg width="30px" height="30px" viewBox="0 0 30 30" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
    <title>Path@1,5x</title>
    <g id="Page-1" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
        <g id="Guide" transform="translate(-81.000000, -942.000000)" fill="#FF0000" fill-rule="nonzero">
            <path d="M108.381187,958.755575 L96.4472189,971.808913 C96.3331604,971.934324 96.1710677,972 96.0030028,972 C95.8349012,972 95.6728451,971.934324 95.5587866,971.808913 L83.6248195,958.755575 C80.1250602,954.933904 80.1250602,948.717756 83.6248195,944.896085 C85.3296508,943.027053 87.6048113,942 90.0300105,942 C92.2511279,942 94.3521779,942.859874 96.003003,944.442257 C97.6538281,942.859874 99.7488704,942 101.969988,942 C104.395187,942 106.676319,943.027053 108.387196,944.896085 C111.874938,948.717756 111.868929,954.933904 108.381187,958.755575 Z" id="Path"></path>
        </g>
    </g>
</svg>
