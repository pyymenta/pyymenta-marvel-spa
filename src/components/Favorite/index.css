.hero-item {
  width: 100%;
  height: 100%;
}

.hero-item__image {
  border-bottom: 5px solid var(--color-primary);
  width: 100%;
}

.hero-item__info-wrapper {
  color: var(--color-typo-strong);
  font-size: 20px;
  font-weight: 600;
  display: flex;
  justify-content: space-between;
  margin-top: 24px;
}
