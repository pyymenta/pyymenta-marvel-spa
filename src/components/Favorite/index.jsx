import React, { useState } from 'react';
import PropTypes from 'prop-types';
import './index.css';
import fullHeart from './full-heart.svg';
import emptyHeart from './empty-heart.svg';

const Favorite = ({ isFavorite, handleFavoritePersistence, itemId }) => {
  const [favorite, setFavorite] = useState(isFavorite);

  const handleFavorite = () => {
    const shouldSetFavorite = handleFavoritePersistence(itemId, !favorite);

    if (!shouldSetFavorite) {
      return;
    }

    setFavorite(!favorite);
  };

  return (
    <img
      className='favorite'
      onClick={handleFavorite}
      src={favorite ? fullHeart : emptyHeart}
      alt='Favoritar ou desfavoritar herÃ³i'
      role='presentation'
    />
  );
};

Favorite.defaultProps = {
  isFavorite: false,
  handleFavoritePersistence: () => true,
  itemId: 0,
};

Favorite.propTypes = {
  isFavorite: PropTypes.bool,
  handleFavoritePersistence: PropTypes.func,
  itemId: PropTypes.number,
};

export default Favorite;
