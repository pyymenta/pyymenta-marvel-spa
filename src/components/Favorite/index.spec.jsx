import React from 'react';
import { shallow } from 'enzyme';
import Favorite from './index';

describe('Favorite component', () => {
  const container = shallow(
    <Favorite isFavorite={false} handleFavoritePersistence={() => true} itemId={1} />
  );

  it('should match the snapshot', () => {
    expect(container.html()).toMatchSnapshot();
  });

  it('should change isFavorite state when component is clicked', () => {
    container.find('.favorite');
    container.find('.favorite').first().simulate('click');

    expect(container.find('.favorite').prop('src')).toContain('full-heart');
  });
});
