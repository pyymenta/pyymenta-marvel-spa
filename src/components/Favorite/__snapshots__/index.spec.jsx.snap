// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorite component should match the snapshot 1`] = `"<img class=\\"favorite\\" src=\\"empty-heart.svg\\" alt=\\"Favoritar ou desfavoritar herÃ³i\\" role=\\"presentation\\"/>"`;
